firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java

```diff
@@ -191,17 +230,45 @@ private void persistFid(String fid) throws FirebaseInstallationsException {
     }
   }
 
-  private Task<String> registerFidIfNecessary(PersistedFidEntry persistedFidEntry) {
+  /**
+   * Registers the FID with FIS servers if FID is in UNREGISTERED state.
+   *
+   * <p>Updates FID registration status to PENDING to avoid multiple network calls to FIS Servers.
+   */
+  private Task<String> registerFidIfNecessary(
+      PersistedFidEntry persistedFidEntry, AwaitListener listener) {
     String fid = persistedFidEntry.getFirebaseInstallationId();
 
     // Check if the fid is unregistered
     if (persistedFidEntry.getRegistrationStatus() == RegistrationStatus.UNREGISTERED) {
```

Comment:
Yet to implement race conditions between multiple getAuthToken calls at the same time.